src/cool_seq_tool/mappers/exon_genomic_coords.py

```diff
@@ -126,42 +127,44 @@ async def transcript_to_genomic_coordinates(
         )
 
         # Ensure valid inputs
+        warnings = []
         if not transcript:
-            return self._return_warnings(resp, "Must provide `transcript`")
-        transcript = transcript.strip()
+            warnings.append("Must provide `transcript`")
+        else:
+            transcript = transcript.strip()
 
         exon_start_exists, exon_end_exists = False, False
         if exon_start is not None:
             if exon_start < 1:
-                return self._return_warnings(resp, "`exon_start` cannot be less than 1")
+                warnings.append("`exon_start` cannot be less than 1")
             exon_start_exists = True
 
         if exon_end is not None:
             if exon_end < 1:
-                return self._return_warnings(resp, "`exon_end` cannot be less than 1")
+                warnings.append("`exon_end` cannot be less than 1")
             exon_end_exists = True
 
         if not exon_start_exists and not exon_end_exists:
-            return self._return_warnings(
-                resp, "Must provide either `exon_start` or `exon_end`"
-            )
+            warnings.append("Must provide either `exon_start` or `exon_end`")
         if exon_start_exists and exon_end_exists and (exon_start > exon_end):
-            return self._return_warnings(
-                resp,
-                f"Start exon {exon_start} is greater than end exon {exon_end}",
+            warnings.append(
+                f"Start exon {exon_start} is greater than end exon {exon_end}"
             )
 
+        if warnings:
+            return self._return_warnings(resp, warnings)
+
         # Get all exons and associated start/end coordinates for transcript
         tx_exons, warning = await self.uta_db.get_tx_exons(transcript)
         if not tx_exons:
-            return self._return_warnings(resp, warning or "")
+            return self._return_warnings(resp, [warning] if warning else [""])
```

Comment:
Why list of empty string here rather than empty list?

Reply:
Not totally sure tbh, was trying to match behavior of what was written already. It seems like the existing code intended to emit a blank warning of some kind as a default? I think empty string makes more sense.